### PR TITLE
Fixed updating of snatch/download status for manual searched episodes.

### PR DESCRIPTION
**Pull request by p0psicles**
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Always make sure the PR has a clear description of why,how, and what your trying to fix. The same goes for improvements or new features. Also if you refering to an issue, make sure a summary is available in the PR.
- [x] Read [contribution guide](https://github.com/PyMedusa/SickRage/blob/master/.github/CONTRIBUTING.md)

The manual snatched episode is now retrieved from the showsList, instead of being build from the class. Because it was newly initiated, there was no reference to the show in memory. Now there is.

### CODE REVIEW

File: sickbeard/webserve.py [hunk 1]
```diff
@@ -1466,7 +1466,7 @@ def pickManualSearch(self, provider=None, rowid=None, manual_search_type='episod
         ep_objs = []
         for episode in episodes:
             if episode:
-                ep_objs.append(TVEpisode(show_obj, int(cached_result['season']), int(episode)))
+                ep_objs.append(show_obj.getEpisode(int(cached_result['season']), int(episode)))
```
Comment (p0psicles) [inline sickbeard/webserve.py]: This was the bi@tch

**Comment by p0psicles** [comment]
https://github.com/pymedusa/SickRage/issues/549

**Comment by p0psicles** [comment]
okay, reported issue by @medariox, now it's not possible anymore to reset the status for a snatched/downloaded episode.

**Comment by p0psicles** [comment]
@labrys whould you be so kind to give this a final look, and merge it? Think is pretty solid.

**Comment by labrys** [comment]
@p0psicles code looks good, haven't tested it yet, but if anyone else has I'm fine with it.

File: sickbeard/manual_search.py [hunk 2]
```diff
@@ -92,14 +92,15 @@ def getEpisodes(search_thread, searchstatus):
...
+                        'episode': ep.episode,
+                        'episodeindexid': ep.indexerid,
+                        'season': ep.season,
                         'searchstatus': searchstatus,
-                        'status': statusStrings[ep_obj.status],
-                        'quality': getQualityClass(ep_obj),
-                        'overview': Overview.overviewStrings[show_obj.getOverview(ep_obj.status)]})
+                        'status': statusStrings[ep.status],
+                        'quality': getQualityClass(ep),
+                        'overview': Overview.overviewStrings[show_obj.getOverview(ep.status)]})
```
Comment (labrys) [inline sickbeard/manual_search.py]: Style change here.  I would do:
```python
results.append({
    'show': show_obj.indexerid,
    'episode': ep.episode,
    'episodeindexid': ep.indexerid,
    'season': ep.season,
    'searchstatus': searchstatus,
    'status': statusStrings[ep.status],
    'quality': getQualityClass(ep),
    'overview': Overview.overviewStrings[show_obj.getOverview(ep.status)],
})
```
That way if the variable `results` changes, indentation of dict keys stays constant.  Also note the `,` after the final key.  It makes it easy to add additional keys later.

Comment (p0psicles) [inline sickbeard/manual_search.py]: I'm for any improvement. But it was already PEP8. If we're going to do it like this, maybe we should provide these kinds of examples in standards & guidelines?

**Comment by p0psicles** [comment]
Approved

**Comment by p0psicles** [comment]
Sorry i was not going to fix 600 QuantifiedCode errors.

**Comment by p0psicles** [comment]
https://github.com/pymedusa/SickRage/issues/392

**Pull request merged**